ge/org.osate.ge.gef.ui/src/org/osate/ge/gef/ui/editor/AgeEditor.java

```diff
@@ -571,8 +576,17 @@ public void refreshColoring(final Collection<DiagramElement> diagramElements) {
 				new DefaultReferenceResolutionService(projectReferenceService));
 		adapterMap.put(IContentOutlinePage.class, outlinePage);
 		adapterMap.put(IPropertySheetPage.class, propertySheetPage);
+
+		// Add preference listener
+		preferences.addPreferenceChangeListener(preferenceChangeListener);
 	}
 
+	private final IPreferenceChangeListener preferenceChangeListener = event -> {
+		if (Objects.equals(event.getKey(), Preferences.SHOW_GRID)) {
+			canvas.setShowGrid(event.getNewValue() == null);
```

Comment:
Why is new value being compared to null?

Reply:
I did think this was weird, it had to do with me setting the value using the preference store.  When the value is set to the default value, it's removed from the preference map instead of just updated.  So when I was setting to "true", it was being removed from the map which resulted in a null value.  I have fixed this.